### PR TITLE
Remove optional braces in function blocks

### PR DESCRIPTION
**Pull request by tgodzik**

### CODE REVIEW

File: scalafmt-tests/src/test/resources/scala3/OptionalBraces_fold.stat [hunk 1]
```diff
@@ -1324,3 +1324,22 @@ val bar = foo({
      x += 1
   arr
 })
+<<< lambda
```
Comment (kitbellew) [inline scalafmt-tests/src/test/resources/scala3/OptionalBraces_fold.stat:1327]: could you please add this test to the 4th file, `Optional Braces.stat`?

**Comment by kitbellew** [comment]
needs rebase

File: scalafmt-tests/src/test/resources/scala3/OptionalBraces_fold.stat (hunk 1 above)
Comment (tgodzik) [reply scalafmt-tests/src/test/resources/scala3/OptionalBraces_fold.stat:1342]: Done!

File: scalafmt-tests/src/test/resources/scala3/OptionalBraces_keep.stat [hunk 2]
```diff
@@ -1407,3 +1407,22 @@ val request: Request = Request(
...
      method = HttpMethod.POST
 )
+<<< lambda
+rewrite.scala3.removeOptionalBraces = yes
+===
+val f: Int => Int = (x: Int) => {
+  x * x
+}
+>>>
+val f: Int => Int = (x: Int) => x * x
```
Comment (kitbellew) [inline scalafmt-tests/src/test/resources/scala3/OptionalBraces_keep.stat:1417]: maybe, in a separate PR, we could fix this. there was a line break in the source, but here it disappeared (and this is `keep`).

**Review by kitbellew — APPROVED** [review]

**Pull request merged**